### ISSUE TITLE
Allow checks to run globally across all resources

### ISSUE THREAD

**Issue by m-wynn** [bridgecrewio/checkov#45]
**Is your feature request related to a problem? Please describe.**
I would like to define global checks, e.g. a Naming Convention check, that can apply globally to all resources, without me having to write out all the supported resources

**Describe the solution you'd like**
I would like to write something like:

```python
from checkov.terraform.checks.resource.base_check import BaseResourceCheck

class NamingCheck(BaseResourceCheck):
    def __init__(self):
        name = "Ensure naming convention is correct"
        id = "MY_GLOBAL_001"
        supported_resources = ['*']
        # ...
```
The registry's get_checks could also look at the `*` resource, de-duplicating as necessary (maybe it could be a set?)

**Describe alternatives you've considered**
Get a list of all terraform resources and put that in the list

**Additional context**
It could also be neat to do, for example, `supported_resources = ["aws_*"]`, or more advanced matching, but that is not necessary for my purposes right now.

"Resource names must start with a letter or underscore, and may contain only letters, digits, underscores, and dashes" ([source](https://www.terraform.io/docs/configuration/resources.html#resource-syntax)) so anything else should be fair game for matching

**Comment by schosterbarak** [comment]
Thank you @m-wynn . Do you see additional use-cases for this feature outside of naming convention?
I wonder if there is a built-in best practice check that we should add to utilize this capability.

**Comment by m-wynn** [comment]
I'll try to think of a few.  Some that we might look at are:
* Avoid hard-coding resource names ARNs
* Add tags on resources
* jsonencode HCL objects intead of writing heredocs with embedded json
* check strings for common misspellings

**Comment by JamesWoolfenden** [comment]
the tags is a good one, as far as AWS json policies shouldn't they all be data sources like data aws_iam_policy_document now?

**Comment by m-wynn** [comment]
They *should* but they syntactically don't have to be (see https://www.terraform.io/docs/providers/aws/r/iam_policy.html )

**Comment by m-wynn** [comment]
There's also other endpoints like https://www.terraform.io/docs/providers/vault/r/generic_endpoint.html which, while non-AWS, could still use jsonencode

**Comment by bobdoah** [comment]
I have a use case to check that all resources _that can be tagged_ are tagged with specific tags. I'm not sure if that could be defined using just wildcards, but it would probably be a start.

**Comment by kmroz0991** [comment]
+1 on this. I have a strong use case for enforcing tags on all resources that can be tagged. This would solve a huge hurdle for my team to ensure naming standards and tagging.

**Comment by nimrodkor** [comment]
Potentially solved in #487 
Leaving this open if you guys have any more insight after the new version  will be released. If you have any more feedback let us know!
